tezos/encoding-derive/src/make.rs

```diff
@@ -0,0 +1,626 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use proc_macro2::Span;
+use syn::spanned::Spanned;
+
+use crate::encoding::*;
+use crate::symbol;
+
+type Result<T> = std::result::Result<T, syn::Error>;
+
+pub fn make_encoding(input: &syn::DeriveInput) -> Result<DataWithEncoding> {
+    let meta = &mut get_encoding_meta(&input.attrs)?;
+    let data_with_encoding = make_data_with_encoding(&input.data, &input.ident, meta)?;
+    Ok(data_with_encoding)
+}
+
+fn make_data_with_encoding<'a>(
+    data: &'a syn::Data,
+    name: &'a syn::Ident,
+    meta: &mut Vec<syn::Meta>,
+) -> Result<DataWithEncoding<'a>> {
+    let encoding = match data {
+        syn::Data::Struct(data_struct) => {
+            Encoding::Struct(make_struct_encoding(data_struct, name, meta)?)
+        }
+        syn::Data::Enum(data_enum) => Encoding::Enum(make_enum_encoding(data_enum, name, meta)?),
+        syn::Data::Union(data_union) => {
+            return Err(error_spanned(
+                data_union.union_token,
+                "`union` is not supported",
+            ))
+        }
+    };
+    let encoding = make_bounded_encoding(meta, encoding)?;
+    assert_empty_meta(meta)?;
+    Ok(DataWithEncoding { name, encoding })
+}
+
+fn make_struct_encoding<'a>(
+    data: &'a syn::DataStruct,
+    name: &'a syn::Ident,
+    _meta: &mut Vec<syn::Meta>,
```

Comment:
`_meta` is not used, it can be removed ?

Reply:
This is kept for having similar signatures for `make_struct_encoding` and `make_enum_encoding`. I agree that it can be removed.